GW Launcher/MulticlientPatch.cs

```diff
@@ -54,26 +55,26 @@ static public GWCAMemory LaunchClient(string path,string args,bool datfix,bool n
             }
 
             IntPtr hThread = IntPtr.Zero;
-            uint dwPID = LaunchClient(path, args, (GWML_FLAGS)((datfix ? 2 : 3) | (nologin ? 4 : 0)), out hThread);
+            uint dwPID = LaunchClient(path, args, (GWML_FLAGS)((int)GWML_FLAGS.KEEP_SUSPENDED | (datfix ? 0 : (int)GWML_FLAGS.NO_DATFIX) | (nologin ? (int)GWML_FLAGS.NO_LOGIN : 0) | (elevated ? (int)GWML_FLAGS.ELEVATED : 0)), out hThread);
             Process proc = Process.GetProcessById((int)dwPID);
             GWCAMemory mem = new GWCAMemory(proc);
-            if (mem.process.Threads[0].ThreadState == ThreadState.Wait && mem.process.Threads[0].WaitReason == ThreadWaitReason.Suspended)
-            {
-                try
-                {
-                    mem.process.Kill();
-                    dwPID = LaunchClient(path, args, (GWML_FLAGS)((datfix ? 2 : 3) | (nologin ? 4 : 0)), out hThread);
-                    proc = Process.GetProcessById((int)dwPID);
-                    mem = new GWCAMemory(proc);
-                }
-                catch (Exception)
-                {
-                    MessageBox.Show("This Guild Wars executable is in a suspended state.\nPlease restart the launcher as admin.", "GWMC - Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    ResumeThread(hThread);
-                    CloseHandle(hThread);
-                    return null;
-                }
-            }
+            //if (mem.process.Threads[0].ThreadState == ThreadState.Wait && mem.process.Threads[0].WaitReason == ThreadWaitReason.Suspended)
+            //{
+            //    try
+            //    {
+            //        mem.process.Kill();
+            //        dwPID = LaunchClient(path, args, (GWML_FLAGS)((datfix ? 2 : 3) | (nologin ? 4 : 0)), out hThread);
+            //        proc = Process.GetProcessById((int)dwPID);
+            //        mem = new GWCAMemory(proc);
+            //    }
+            //    catch (Exception)
+            //    {
+            //        MessageBox.Show("This Guild Wars executable is in a suspended state.\nPlease restart the launcher as admin.", "GWMC - Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            //        ResumeThread(hThread);
+            //        CloseHandle(hThread);
+            //        return null;
+            //    }
+            //}

```

Comment:
This will always trigger given that the process is still suspended at this point. I haven't implemented any checks but removed the (feel free to correct me) useless check.